src/multibody/geometry.hxx

```diff
@@ -191,18 +205,17 @@ namespace se3
     activeCollisionPairs[pairId] = false;
   }
 
-  inline CollisionResult GeometryData::computeCollision(const CollisionPair & pair) const
+  inline bool GeometryData::computeCollision(const CollisionPair & pair, fcl::CollisionResult& collisionResult) const
   {
     const Index & co1 = pair.first;     assert(co1<collisionObjects.size());
     const Index & co2 = pair.second;    assert(co2<collisionObjects.size());
 
-    fcl::CollisionRequest collisionRequest (1, false, false, 1, false, true, fcl::GST_INDEP);
-    fcl::CollisionResult collisionResult;
+    static const fcl::CollisionRequest collisionRequest (1, false, false, 1, false, true, fcl::GST_INDEP);
```

Comment:
My concern is not pre-allocation as `const` or `static const` are pre-allocated anyway. The goal is to give to the user the ability to change the request. For instance, one may want to enable contact points computation. I guess @stonneau uses this FCL feature.

I do not see why `static` would be a problem when parallelizing the calling algo.

From here, I'm not sure of my answer. Here is how I see this question of static vs non-static:

I think the main difference between the `static const` and `const` is the place where there are some space allocated to the object.
With `const`, some memory is allocated in the function stack. With `static const`, the memory is allocated within the instruction set. For small objects, I guess allocating in the function stack does not hurt.